### PR TITLE
[CLOB-1050] update daemon liquidation info to include negative tnc su…

### PR DESCRIPTION
**Pull request by jayy04**
…baccounts and open positions

### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/daemons/server/types/liquidations/daemon_liquidation_info.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+)
+
+// DaemonLiquidationInfo maintains the list of subaccount ids to be liquidated
+// in the next block. Methods are goroutine safe.
+type DaemonLiquidationInfo struct {
+	sync.Mutex                                       // lock
+	blockHeight               uint32                 // block height of the last update
+	liquidatableSubaccountIds []satypes.SubaccountId // liquidatable subaccount ids
+	negativeTncSubaccountIds  []satypes.SubaccountId // negative total net collateral subaccount ids
+	subaccountsWithPositions  map[uint32]clobtypes.SubaccountOpenPositionInfo
```
Comment (jayy04) [inline protocol/daemons/server/types/liquidations/daemon_liquidation_info.go:17]: @jakob-dydx I thought `map[uint32]map[bool]map[subaccountId]struct{}` was kind of messy so I created a struct called `SubaccountOpenPositionInfo`. let me know what you think

Comment (lucas-dydx) [inline protocol/daemons/server/types/liquidations/daemon_liquidation_info.go:17]: Suggestion: less error prone to have values of `subaccountsWithPositions` be a pointer? Cause default value of `clobtypes.SubaccountOpenPositionInfo` is technically valid (perp ID of 0, no open positions)

File: protocol/daemons/server/types/liquidations/daemon_liquidation_info.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+)
+
+// DaemonLiquidationInfo maintains the list of subaccount ids to be liquidated
+// in the next block. Methods are goroutine safe.
+type DaemonLiquidationInfo struct {
+	sync.Mutex                                       // lock
+	blockHeight               uint32                 // block height of the last update
```
Comment (lucas-dydx) [inline protocol/daemons/server/types/liquidations/daemon_liquidation_info.go:14]: Are all of the liquidatable / negative TNC subaccounts read from the same block height (which is the same as this `blockHeight` field)?

**Review by lucas-dydx — APPROVED** [review]

**Review by jakob-dydx — APPROVED** [review]

File: protocol/daemons/server/types/liquidations/daemon_liquidation_info.go (hunk 2 above)
Comment (jayy04) [reply protocol/daemons/server/types/liquidations/daemon_liquidation_info.go:14]: yes - I will be updating the liquidation daemon to make height based queries so everything is read from the same height

**Pull request merged**